private/buf/bufgen/internal/bufgenv2/managed_mode.go

```diff
@@ -43,34 +41,77 @@ func applyManagementForFile(
 ) error {
 	for _, fileOption := range AllFileOptions {
 		if managedConfig.DisabledFunc(fileOption, imageFile) {
+			// disable has a higher precedence over override
 			continue
 		}
 		var override bufimagemodifyv2.Override
 		if overrideFunc, ok := managedConfig.FileOptionToOverrideFunc[fileOption]; ok {
 			override = overrideFunc(imageFile)
 		}
-		// override can be nil at this point, in which case ModifyXYZ will either use
-		// our default value for the option, or leave the option alone, implicitly
-		// using Protobuf's default value.
+		// override can be nil at this point, which is a valid argument to ModifyXYZ
 		switch fileOption {
```

Comment:
perhaps having a `map[fileOption]ModifyFunc` is better here, but that could potentially make the code a little bit (just a little) harder to read.